File: objectModel/Python/cdm/persistence/cdmfolder/types/e2e_relationship.py
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License. See License.txt in the project root for license information.

from cdm.utilities import JObject
from typing import Union, List


class E2ERelationship(JObject):
    def __init__(self):
        super().__init__()

        self.name = None  # type: str
        self.fromEntity = None  # type: str
        self.fromEntityAttribute = None  # type: str
        self.toEntity = None  # type: str
        self.toEntityAttribute = None  # type: str
        self.exhibitsTraits = None  # type: List[Union[str, TraitReference]]

File: objectModel/TypeScript/Persistence/CdmFolder/types/E2ERelationship.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

import {
    TraitGroupReference,
    TraitReference
} from '.';

export abstract class E2ERelationship {
    public name: string;
    public fromEntity: string;
    public fromEntityAttribute: string;
    public toEntity: string;
    public toEntityAttribute: string;
    public exhibitsTraits?: (string | TraitReference | TraitGroupReference)[];
}
